src/Psl/Option/Exception/NoneException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Option\Exception;
+
+use Psl\Exception\UnderflowException;
+
+final class NoneException extends UnderflowException implements ExceptionInterface
```

Comment:
I don't like the exception name, not sure what else would work.

Reply:
`UnwrapException` / `UnwrapNoneException` ?

Reply:
Keeping this one for now seems ok.